app/src/main/java/cloudcity/LoggingServiceExtensions.java

```diff
@@ -146,6 +151,10 @@ public static void stopCloudCity() {
     }
 
     private static NetworkDataModel getCloudCityData() {
+        if (dp == null) {
+            Log.e(TAG, "DataProvider was null! Bailing out, returning null...");
+            return null;
+        }
```

Comment:
we could have guarenteed that if we were doing Kotlin, but since we're not doing Kotlin we cannot guarantee that something just wont become null because <random reasons>. While the concern is valid, there's no point in implementing it as it can't solve the issue at hand.

The annotation doesn't prevent us from passing nulls into the method at run-time, it only catches null arguments at compile-time
The null-check does prevent us from going further and incurring a NullPointerException if a null does happen to get passed into the method.

resolving.

Reply:
also please learn from these comments, i feel like this is the 5th time i'm telling you this exact same thing. annotations are useful during compile-time, java doesn't have nullability as a language feature like kotlin hence it cannot prevent things from becoming null.